### PR TITLE
[Clustering] Enable RNN layers

### PR DESCRIPTION
**Pull request by johan-gras**
This PR enables the support of RNN layers in clustering.

### State of support
Here is the current state of support for RNN:
| Type of RNN      | Status |
| ----------- | ----------- |
| RNNSimple      | ✅       |
| LSTM      | ✅       |
| GRU      | ✅       |
| PeepholeLSTM      | ❌       |
| BidirectionalRNN | ❌       |
| StackedRNN | ❌       |

### Sanity check results
To confirm that the implementation seems robust enough, we ran some experiments on the IMDB dataset and compared accuracies to the existing pruning implementation with a LSTM layer.

**Setup**: Direct pruning/clustering from random weight (no finetuning) for 3 epochs on the IMDB dataset. 0.7% sparsity for pruning or 16 clusters for clustering.
| Type of RNN      | Accuracy |
| ----------- | ----------- |
| **LSTM (pruning baseline)**      | **0.8485** |
| LSTM (clustering)     | 0.8468       |
| GRU (clustering)     | 0.8536 |
| SimpleRNN (clustering) | 0.7690      |

### CODE REVIEW

**Review by wwwind — APPROVED** [review]

File: tensorflow_model_optimization/python/core/clustering/keras/cluster.py [hunk 1]
```diff
@@ -236,6 +237,14 @@ def _add_clustering_wrapper(layer):
       return layer
     if isinstance(layer, InputLayer):
       return layer.__class__.from_config(layer.get_config())
+    if isinstance(layer, keras.layers.RNN):
```
[the remark this answers is not in the document]
Comment (johan-gras) [reply tensorflow_model_optimization/python/core/clustering/keras/cluster.py]: Thank you, done.

**Review by fredrec — APPROVED** [review]
